src/DotNetWorker.ApplicationInsights/Initializers/FunctionsApplicationInsightsOptionsInitializer.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker.ApplicationInsights.Initializers
+{
+    internal class FunctionsApplicationInsightsOptionsInitializer : IConfigureOptions<FunctionsApplicationInsightsOptions>
+    {
+        private const string AppInsightsAuthenticationString = "APPLICATIONINSIGHTS_AUTHENTICATION_STRING";
+
+        public void Configure(FunctionsApplicationInsightsOptions options)
+        {
+            // Read AuthString from environment variable
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(AppInsightsAuthenticationString)))
```

Comment:
Before reading environment, check if the same value is in `IConfiguration`, then fallback to reading environment directly.

Reply:
Done.